hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java

```diff
@@ -750,6 +750,49 @@
             @Nonnull BiFunctionEx<T, T1, R> mapToOutputFn
     );
 
+    /**
+     * Attaches to both this and the supplied stage an inner hash-joining stage
+     * and  returns it. This stage plays the role of the <em>primary stage</em>
+     * in the hash-join. Please refer to the {@link com.hazelcast.jet.pipeline
+     * package javadoc} for a detailed description of the hash-join transform.
+     * <p>
+     * This sample joins a stream of users to a stream of countries and outputs
+     * a stream of users with the {@code country} field set:
+     * <pre>{@code
+     * // Types of the input stages:
+     * BatchStage<User> users;
+     * BatchStage<Map.Entry<Long, Country>> idAndCountry;
+     *
+     * users.innerHashJoin(
+     *     idAndCountry,
+     *     JoinClause.joinMapEntries(User::getCountryId),
+     *     (user, country) -> user.setCountry(country)
+     * )
+     * }</pre>
+     *
+     * <p>
+     * This method is similar to {@link #hashJoin(BatchStage, JoinClause, BiFunctionEx)}
```

Comment:
Line too long. Note that you can just refer to a method name without parameters.